src/object/cli_shard.c

```diff
@@ -300,6 +302,22 @@ dc_rw_cb(tse_task_t *task, void *arg)
 		D_GOTO(out, ret);
 	}
 
+	/*
+	 * orwo->orw_epoch may be set even when the status is nonzero (e.g.,
+	 * -DER_TX_RESTART and -DER_INPROGRESS). If this task is a TX "epoch
+	 * task", we shall call dc_tx_set_epoch.
```

Comment:
Is is possible that the orw_epoch is not initialized if the returned RPC status is non-zero?

Reply:
Uninitialized orw_epoch == 0.